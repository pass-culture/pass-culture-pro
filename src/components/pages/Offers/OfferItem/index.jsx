import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { compose } from 'redux'

import RawOfferItem from './RawOfferItem'
import selectEventById from 'selectors/selectEventById'
import selectAggregatedStockByOfferId from 'selectors/selectAggregatedStockByOfferId'
import selectMaxDateByOfferId from 'selectors/selectMaxDateByOfferId'
import selectMediationsByOfferId from 'selectors/selectMediationsByOfferId'
import selectStocksByOfferId from 'selectors/selectStocksByOfferId'
import selectThingById from 'selectors/selectThingById'
import selectThumbUrlByOfferIdAndEventIdAndThingId from 'selectors/selectThumbUrlByOfferIdAndEventIdAndThingId'
import selectVenueById from 'selectors/selectVenueById'
import offererSelector from 'selectors/selectOffererById'
import { getOfferTypeLabel } from 'utils/offerItem'

function mapStateToProps(state, ownProps) {
  const { offer } = ownProps
  const { id: offerId, eventId, thingId, venueId } = offer
  const event = selectEventById(state, eventId)
  const thing = selectThingById(state, thingId)
  const venue = selectVenueById(state, venueId)
  const offerer = offererSelector(state, venue.managingOffererId)
  const stockAlertMessage = offer.stockAlertMessage

  return {
    aggregatedStock: selectAggregatedStockByOfferId(state, offerId),
    event,
    maxDate: selectMaxDateByOfferId(state, offerId),
    mediations: selectMediationsByOfferId(state, offerId),
    stocks: selectStocksByOfferId(state, offerId),
    thing,
    thumbUrl: selectThumbUrlByOfferIdAndEventIdAndThingId(
      state,
      offerId,
      eventId,
      thingId
    ),
    offerer,
    offerTypeLabel: getOfferTypeLabel(event, thing),
    stockAlertMessage,
    venue,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(RawOfferItem)
