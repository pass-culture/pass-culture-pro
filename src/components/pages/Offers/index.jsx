import { lastTrackerMoment } from 'pass-culture-shared'
import { connect } from 'react-redux'
import { compose } from 'redux'
import withQueryRouter from 'with-query-router'

import RawOffers from './RawOffers'
import { withRedirectToSigninWhenNotAuthenticated } from 'components/hocs'
import selectOffererById from 'selectors/selectOffererById'
import selectOffersByOffererIdAndVenueId from 'selectors/selectOffersByOffererIdAndVenueId'
import selectVenueById from 'selectors/selectVenueById'
import { translateQueryParamsToApiParams } from 'utils/translate'

export function mapStateToProps(state, ownProps) {
  const { query } = ownProps
  const queryParams = query.parse()
  const apiQueryParams = translateQueryParamsToApiParams(queryParams)
  const { offererId, venueId } = apiQueryParams
  return {
    lastTrackerMoment: lastTrackerMoment(state, 'offers'),
    offers: selectOffersByOffererIdAndVenueId(state, offererId, venueId),
    offerer: selectOffererById(state, offererId),
    types: state.data.types,
    venue: selectVenueById(state, venueId),
  }
}

export default compose(
  withRedirectToSigninWhenNotAuthenticated,
  withQueryRouter,
  connect(mapStateToProps)
)(RawOffers)
