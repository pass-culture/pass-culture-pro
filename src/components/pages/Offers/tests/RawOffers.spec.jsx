import React from 'react'
import { shallow } from 'enzyme'
import { NavLink } from 'react-router-dom'

import Offers from '../Offers'
import OfferItem from '../OfferItem/OfferItemContainer'
import mockedOffers from './offersMock'

const dispatchMock = jest.fn()
const queryChangeMock = jest.fn()
const initialProps = {
  currentUser: {},
  dispatch: dispatchMock,
  offers: [],
  location: {
    pathname: '/offres',
  },
  pagination: {
    apiQuery: {
      keywords: null,
      offererId: null,
      orderBy: 'offer.id+desc',
      venueId: null,
    },
  },
  query: {
    change: queryChangeMock,
    parse: () => ({ orderBy: 'offer.id desc' }),
  },
  search: '',
  types: [],
}

describe('src | components | pages | Offers | Offers', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // when
      const wrapper = shallow(<Offers {...initialProps} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
      dispatchMock.mockClear()
      queryChangeMock.mockClear()
    })
  })

  describe('render', () => {
    describe('when arriving on index page', () => {
      it('should list all offers', () => {
        // given
        const expected = {
          hasMore: false,
          isLoading: true,
        }

        // when
        const wrapper = shallow(<Offers {...initialProps} />)

        // then
        expect(wrapper.state()).toStrictEqual(expected)
        dispatchMock.mockClear()
        queryChangeMock.mockClear()
      })
    })
    describe('offerItem', () => {
      it('should render items corresponding to offers', () => {
        // given
        initialProps.offers = mockedOffers

        // when
        const wrapper = shallow(<Offers {...initialProps} />)
        const offerItem = wrapper.find(OfferItem)

        // then
        expect(offerItem).toHaveLength(mockedOffers.length)
      })
    })
    describe('when offers shown for an offerer or a venue', () => {
      describe('navLink to create offer', () => {
        describe('when user isAdmin', () => {
          it('should display NavLink', () => {
            // given
            initialProps.currentUser = {
              isAdmin: true,
            }

            // when
            const wrapper = shallow(<Offers {...initialProps} />)
            const navLink = wrapper.find(NavLink)

            // then
            expect(navLink).toHaveLength(0)
          })
        })
        describe('when structure (or offererId)', () => {
          it('should render link properly', () => {
            // given
            initialProps.currentUser = {
              isAdmin: false,
            }
            const parseMock = () => ({ structure: 'TEST' })
            initialProps.query.parse = parseMock

            // when
            const wrapper = shallow(<Offers {...initialProps} />)

            const navLink = wrapper.find(NavLink)

            // then
            expect(navLink).toHaveLength(1)
            expect(navLink.props().to).toStrictEqual('/offres/creation?structure=TEST')
          })
        })
        describe('when lieu or (VenueId)', () => {
          it('should render link properly', () => {
            // given
            initialProps.currentUser = {
              isAdmin: false,
            }
            const parseMock = () => ({ lieu: 'LIEU' })
            initialProps.query.parse = parseMock

            // when
            const wrapper = shallow(<Offers {...initialProps} />)

            const navLink = wrapper.find(NavLink)

            // then
            expect(navLink).toHaveLength(1)
            expect(navLink.props().to).toStrictEqual('/offres/creation?lieu=LIEU')
          })
        })
      })
    })
  })

  describe('functions', () => {
    describe('constructor', () => {
      it('should dispatch assignData when component is constructed', () => {
        // when
        shallow(<Offers {...initialProps} />)
        const expectedAssignData = {
          patch: {
            offers: [],
          },
          type: 'ASSIGN_DATA',
        }

        // then
        expect(dispatchMock.mock.calls[0][0]).toStrictEqual(expectedAssignData)
        dispatchMock.mockClear()
        queryChangeMock.mockClear()
      })
    })

    describe('componentDidMount', () => {
      describe('when there is no pagination', () => {
        it('should dispatch handle RequestData', () => {
          // given
          const parseMock = () => ({
            keywords: 'océan',
            orderBy: 'offer.id desc',
          })
          initialProps.query.parse = parseMock

          // when
          const wrapper = shallow(<Offers {...initialProps} />)
          wrapper.instance().componentDidMount()
          const expected = {
            config: {
              apiPath: '/types',
              method: 'GET',
            },
            type: 'REQUEST_DATA_GET_/TYPES',
          }

          // then
          expect(dispatchMock.mock.calls[1][0]).toStrictEqual(expected)
          dispatchMock.mockClear()
        })
      })
    })

    describe('componentDidUpdate', () => {
      describe('when location has change', () => {
        it('should dispatch assignData when component is rendered', () => {
          // given
          const prevProps = initialProps

          initialProps.location = {
            pathname: '/offres',
            search: '?orderBy=offer.id+desc',
          }

          // when
          const wrapper = shallow(<Offers {...initialProps} />)
          wrapper.instance().componentDidUpdate(prevProps)

          const expectedAssignData = {
            config: {
              apiPath: '/types',
              method: 'GET',
            },
            type: 'REQUEST_DATA_GET_/TYPES',
          }

          // the
          expect(dispatchMock.mock.calls[1][0]).toStrictEqual(expectedAssignData)

          dispatchMock.mockClear()
        })
      })
    })

    describe('handleOnSubmit', () => {
      // when
      const wrapper = shallow(<Offers {...initialProps} />)
      const event = Object.assign(jest.fn(), {
        preventDefault: () => {},
        target: {
          elements: {
            search: {
              value: 'AnyWord',
            },
          },
        },
      })

      describe('when keywords is not an empty string', () => {
        it('should update state with isLoading to true', () => {
          // when
          wrapper.instance().handleOnSubmit(event)
          const expected = {
            hasMore: false,
            isLoading: true,
          }

          // then
          expect(wrapper.state()).toStrictEqual(expected)
        })

        it('should dispatch assignData with good params', () => {
          // when
          wrapper.instance().handleOnSubmit(event)

          const expected = {
            patch: {
              offers: [],
            },
            type: 'ASSIGN_DATA',
          }

          // then
          expect(dispatchMock.mock.calls[0][0]).toStrictEqual(expected)
          dispatchMock.mockClear()
        })

        it('should change query', () => {
          // when
          wrapper.instance().handleOnSubmit(event)
          const expected = {
            'mots-cles': 'AnyWord',
            page: null,
          }

          // then
          expect(queryChangeMock.mock.calls[0][0]).toStrictEqual(expected)
          queryChangeMock.mockClear()
        })
      })

      describe('when keywords is an empty string', () => {
        it('should change query with mots-clés setted to null', () => {
          // given
          const eventEmptyWord = Object.assign(jest.fn(), {
            preventDefault: () => {},
            target: {
              elements: {
                search: {
                  value: '',
                },
              },
            },
          })

          // when
          wrapper.instance().handleOnSubmit(eventEmptyWord)
          const expected = {
            'mots-cles': null,
            page: null,
          }

          // then
          expect(queryChangeMock.mock.calls[0][0]).toStrictEqual(expected)
          queryChangeMock.mockClear()
        })
      })
    })
  })
})
