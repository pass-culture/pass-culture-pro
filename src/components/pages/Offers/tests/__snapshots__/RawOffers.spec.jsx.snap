// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Offers | Offers snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Offers
    currentUser={Object {}}
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "patch": Object {
                "offers": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {
                "apiPath": "/types",
                "method": "GET",
              },
              "type": "REQUEST_DATA_GET_/TYPES",
            },
          ],
          Array [
            Object {
              "config": Object {
                "apiPath": "/offers?orderBy=offer.id%20desc",
                "handleFail": [Function],
                "handleSuccess": [Function],
                "method": "GET",
                "normalizer": Object {
                  "mediations": "mediations",
                  "product": Object {
                    "normalizer": Object {
                      "offers": "offers",
                    },
                    "stateKey": "products",
                  },
                  "stocks": "stocks",
                  "venue": Object {
                    "normalizer": Object {
                      "managingOfferer": "offerers",
                    },
                    "stateKey": "venues",
                  },
                },
                "resolve": [Function],
              },
              "type": "REQUEST_DATA_GET_/OFFERS?ORDERBY=OFFER.ID%20DESC",
            },
          ],
          Array [
            Object {
              "patch": Object {
                "offers": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {
                "apiPath": "/types",
                "method": "GET",
              },
              "type": "REQUEST_DATA_GET_/TYPES",
            },
          ],
          Array [
            Object {
              "config": Object {
                "apiPath": "/offers?orderBy=offer.id%20desc",
                "handleFail": [Function],
                "handleSuccess": [Function],
                "method": "GET",
                "normalizer": Object {
                  "mediations": "mediations",
                  "product": Object {
                    "normalizer": Object {
                      "offers": "offers",
                    },
                    "stateKey": "products",
                  },
                  "stocks": "stocks",
                  "venue": Object {
                    "normalizer": Object {
                      "managingOfferer": "offerers",
                    },
                    "stateKey": "venues",
                  },
                },
                "resolve": [Function],
              },
              "type": "REQUEST_DATA_GET_/OFFERS?ORDERBY=OFFER.ID%20DESC",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    location={
      Object {
        "pathname": "/offres",
      }
    }
    offers={Array []}
    pagination={
      Object {
        "apiQuery": Object {
          "keywords": null,
          "offererId": null,
          "orderBy": "offer.id+desc",
          "venueId": null,
        },
      }
    }
    query={
      Object {
        "change": [MockFunction],
        "parse": [Function],
      }
    }
    search=""
    types={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <HeroSection
          subtitle={null}
          title="Vos offres"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="cta button is-primary"
            to="/offres/creation"
          >
            <span
              className="icon"
            >
              <Icon
                svg="ico-offres-w"
              />
            </span>
            <span>
              Créer une offre
            </span>
          </NavLink>
        </HeroSection>,
        <form
          className="section"
          onSubmit={[Function]}
        >
          <label
            className="label"
          >
            Rechercher une offre :
          </label>
          <div
            className="field is-grouped"
          >
            <p
              className="control is-expanded"
            >
              <input
                className="input"
                id="search"
                placeholder="Saisissez un ou plusieurs mots complets"
                type="text"
              />
            </p>
            <p
              className="control"
            >
              <button
                className="button is-primary is-outlined"
                id="search-button"
                type="submit"
              >
                OK
              </button>
              <button
                className="button is-secondary"
                disabled={true}
                type="button"
              >
                   
                <Icon
                  svg="ico-filter"
                />
              </button>
            </p>
          </div>
        </form>,
        <ul
          className="section"
        />,
        <div
          className="section"
        >
          <withRouter(_withQueryRouter)
            className="offers-list main-list"
            element="ul"
            hasMore={false}
            isLoading={true}
            loader={
              <Spinner
                Tag="div"
                className={null}
                dotFrequency={500}
                label="Chargement"
                style={null}
              />
            }
            useWindow={true}
          />
        </div>,
      ],
      "handleRequestData": [Function],
      "name": "offers",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            className="cta button is-primary"
            to="/offres/creation"
          >
            <span
              className="icon"
            >
              <Icon
                svg="ico-offres-w"
              />
            </span>
            <span>
              Créer une offre
            </span>
          </NavLink>,
          "subtitle": null,
          "title": "Vos offres",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": Array [
              <span
                className="icon"
              >
                <Icon
                  svg="ico-offres-w"
                />
              </span>,
              <span>
                Créer une offre
              </span>,
            ],
            "className": "cta button is-primary",
            "to": "/offres/creation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  svg="ico-offres-w"
                />,
                "className": "icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "svg": "ico-offres-w",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Créer une offre",
              },
              "ref": null,
              "rendered": "Créer une offre",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="label"
            >
              Rechercher une offre :
            </label>,
            <div
              className="field is-grouped"
            >
              <p
                className="control is-expanded"
              >
                <input
                  className="input"
                  id="search"
                  placeholder="Saisissez un ou plusieurs mots complets"
                  type="text"
                />
              </p>
              <p
                className="control"
              >
                <button
                  className="button is-primary is-outlined"
                  id="search-button"
                  type="submit"
                >
                  OK
                </button>
                <button
                  className="button is-secondary"
                  disabled={true}
                  type="button"
                >
                     
                  <Icon
                    svg="ico-filter"
                  />
                </button>
              </p>
            </div>,
          ],
          "className": "section",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rechercher une offre :",
              "className": "label",
            },
            "ref": null,
            "rendered": "Rechercher une offre :",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="control is-expanded"
                >
                  <input
                    className="input"
                    id="search"
                    placeholder="Saisissez un ou plusieurs mots complets"
                    type="text"
                  />
                </p>,
                <p
                  className="control"
                >
                  <button
                    className="button is-primary is-outlined"
                    id="search-button"
                    type="submit"
                  >
                    OK
                  </button>
                  <button
                    className="button is-secondary"
                    disabled={true}
                    type="button"
                  >
                       
                    <Icon
                      svg="ico-filter"
                    />
                  </button>
                </p>,
              ],
              "className": "field is-grouped",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    className="input"
                    id="search"
                    placeholder="Saisissez un ou plusieurs mots complets"
                    type="text"
                  />,
                  "className": "control is-expanded",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input",
                    "defaultValue": undefined,
                    "id": "search",
                    "placeholder": "Saisissez un ou plusieurs mots complets",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="button is-primary is-outlined"
                      id="search-button"
                      type="submit"
                    >
                      OK
                    </button>,
                    <button
                      className="button is-secondary"
                      disabled={true}
                      type="button"
                    >
                         
                      <Icon
                        svg="ico-filter"
                      />
                    </button>,
                  ],
                  "className": "control",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "OK",
                      "className": "button is-primary is-outlined",
                      "id": "search-button",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "OK",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "   ",
                        <Icon
                          svg="ico-filter"
                        />,
                      ],
                      "className": "button is-secondary",
                      "disabled": true,
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Array [
                      "   ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "svg": "ico-filter",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "button",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "section",
        },
        "ref": null,
        "rendered": null,
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <withRouter(_withQueryRouter)
              className="offers-list main-list"
              element="ul"
              hasMore={false}
              isLoading={true}
              loader={
                <Spinner
                  Tag="div"
                  className={null}
                  dotFrequency={500}
                  label="Chargement"
                  style={null}
                />
              }
              useWindow={true}
            />,
          ],
          "className": "section",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [],
              "className": "offers-list main-list",
              "element": "ul",
              "hasMore": false,
              "isLoading": true,
              "loader": <Spinner
                Tag="div"
                className={null}
                dotFrequency={500}
                label="Chargement"
                style={null}
              />,
              "useWindow": true,
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <HeroSection
            subtitle={null}
            title="Vos offres"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="cta button is-primary"
              to="/offres/creation"
            >
              <span
                className="icon"
              >
                <Icon
                  svg="ico-offres-w"
                />
              </span>
              <span>
                Créer une offre
              </span>
            </NavLink>
          </HeroSection>,
          <form
            className="section"
            onSubmit={[Function]}
          >
            <label
              className="label"
            >
              Rechercher une offre :
            </label>
            <div
              className="field is-grouped"
            >
              <p
                className="control is-expanded"
              >
                <input
                  className="input"
                  id="search"
                  placeholder="Saisissez un ou plusieurs mots complets"
                  type="text"
                />
              </p>
              <p
                className="control"
              >
                <button
                  className="button is-primary is-outlined"
                  id="search-button"
                  type="submit"
                >
                  OK
                </button>
                <button
                  className="button is-secondary"
                  disabled={true}
                  type="button"
                >
                     
                  <Icon
                    svg="ico-filter"
                  />
                </button>
              </p>
            </div>
          </form>,
          <ul
            className="section"
          />,
          <div
            className="section"
          >
            <withRouter(_withQueryRouter)
              className="offers-list main-list"
              element="ul"
              hasMore={false}
              isLoading={true}
              loader={
                <Spinner
                  Tag="div"
                  className={null}
                  dotFrequency={500}
                  label="Chargement"
                  style={null}
                />
              }
              useWindow={true}
            />
          </div>,
        ],
        "handleRequestData": [Function],
        "name": "offers",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="cta button is-primary"
              to="/offres/creation"
            >
              <span
                className="icon"
              >
                <Icon
                  svg="ico-offres-w"
                />
              </span>
              <span>
                Créer une offre
              </span>
            </NavLink>,
            "subtitle": null,
            "title": "Vos offres",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": Array [
                <span
                  className="icon"
                >
                  <Icon
                    svg="ico-offres-w"
                  />
                </span>,
                <span>
                  Créer une offre
                </span>,
              ],
              "className": "cta button is-primary",
              "to": "/offres/creation",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Icon
                    svg="ico-offres-w"
                  />,
                  "className": "icon",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "svg": "ico-offres-w",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Créer une offre",
                },
                "ref": null,
                "rendered": "Créer une offre",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="label"
              >
                Rechercher une offre :
              </label>,
              <div
                className="field is-grouped"
              >
                <p
                  className="control is-expanded"
                >
                  <input
                    className="input"
                    id="search"
                    placeholder="Saisissez un ou plusieurs mots complets"
                    type="text"
                  />
                </p>
                <p
                  className="control"
                >
                  <button
                    className="button is-primary is-outlined"
                    id="search-button"
                    type="submit"
                  >
                    OK
                  </button>
                  <button
                    className="button is-secondary"
                    disabled={true}
                    type="button"
                  >
                       
                    <Icon
                      svg="ico-filter"
                    />
                  </button>
                </p>
              </div>,
            ],
            "className": "section",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rechercher une offre :",
                "className": "label",
              },
              "ref": null,
              "rendered": "Rechercher une offre :",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="control is-expanded"
                  >
                    <input
                      className="input"
                      id="search"
                      placeholder="Saisissez un ou plusieurs mots complets"
                      type="text"
                    />
                  </p>,
                  <p
                    className="control"
                  >
                    <button
                      className="button is-primary is-outlined"
                      id="search-button"
                      type="submit"
                    >
                      OK
                    </button>
                    <button
                      className="button is-secondary"
                      disabled={true}
                      type="button"
                    >
                         
                      <Icon
                        svg="ico-filter"
                      />
                    </button>
                  </p>,
                ],
                "className": "field is-grouped",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <input
                      className="input"
                      id="search"
                      placeholder="Saisissez un ou plusieurs mots complets"
                      type="text"
                    />,
                    "className": "control is-expanded",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "input",
                      "defaultValue": undefined,
                      "id": "search",
                      "placeholder": "Saisissez un ou plusieurs mots complets",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <button
                        className="button is-primary is-outlined"
                        id="search-button"
                        type="submit"
                      >
                        OK
                      </button>,
                      <button
                        className="button is-secondary"
                        disabled={true}
                        type="button"
                      >
                           
                        <Icon
                          svg="ico-filter"
                        />
                      </button>,
                    ],
                    "className": "control",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "OK",
                        "className": "button is-primary is-outlined",
                        "id": "search-button",
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": "OK",
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "   ",
                          <Icon
                            svg="ico-filter"
                          />,
                        ],
                        "className": "button is-secondary",
                        "disabled": true,
                        "type": "button",
                      },
                      "ref": null,
                      "rendered": Array [
                        "   ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "svg": "ico-filter",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                      ],
                      "type": "button",
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "section",
          },
          "ref": null,
          "rendered": null,
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <withRouter(_withQueryRouter)
                className="offers-list main-list"
                element="ul"
                hasMore={false}
                isLoading={true}
                loader={
                  <Spinner
                    Tag="div"
                    className={null}
                    dotFrequency={500}
                    label="Chargement"
                    style={null}
                  />
                }
                useWindow={true}
              />,
            ],
            "className": "section",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [],
                "className": "offers-list main-list",
                "element": "ul",
                "hasMore": false,
                "isLoading": true,
                "loader": <Spinner
                  Tag="div"
                  className={null}
                  dotFrequency={500}
                  label="Chargement"
                  style={null}
                />,
                "useWindow": true,
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
